### PR TITLE
Image upload

### PR DESCRIPTION
**Pull request by weili-govtech**
**Changes**
- add support for image-upload field
- delete branch

**Changelog entry**
- Add `image-upload` field

### CODE REVIEW

File: src/utils/file-helper.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import { fromBuffer } from "file-type";
+
+export namespace FileHelper {
+	/**
+	 * formats bytes to bigger units
+	 */
+	export const bytesToSize = (bytes: number) => {
```
Comment (qroll) [inline src/utils/file-helper.ts:7]: this util is not needed on the backend and can be removed as well

(p.s. I just noticed it's also unused in FEE, was it intended to format the file size display?)

**Review by qroll — CHANGES_REQUESTED** [review]

File: src/fields/image-upload.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+							!value ||
+							!Array.isArray(value) ||
+							!compress ||
+							!dimensions ||
+							!dimensions.width ||
+							!dimensions.height
+						)
+							return true;
+
+						return value.reduce((accumulator, file) => {
```
[the remark this answers is not in the document]
Comment (weili-govtech) [reply src/fields/image-upload.ts]: done

File: src/fields/image-upload.ts [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+type TImageUploadAcceptedFileType = "jpg" | "gif" | "png" | "heic" | "heif" | "webp";
+type TImageUploadOutputFileType = "jpg" | "png";
+interface IImageDimensions {
+	width: number;
+	height: number;
+}
+
+interface IImageUploadValidationRule extends IValidationRule {
+	compress?: boolean | undefined;
```
[the remark this answers is not in the document]
Comment (weili-govtech) [reply src/fields/image-upload.ts]: seems like I added wrongly, have removed

File: src/utils/file-helper.ts (hunk 1 above)
Comment (weili-govtech) [reply src/utils/file-helper.ts]: yeah it was intended to render the file size but it appears not to be used so I have removed it in both repos

**Review by qroll — APPROVED** [review]

**Pull request merged**